.github/actions/github-release/main.js

```diff
@@ -77,39 +77,47 @@ async function runOnce() {
   // Upload all the relevant assets for this release as just general blobs.
   for (const file of glob.sync(files)) {
     const size = fs.statSync(file).size;
-    core.info(`upload ${file}`);
-    await octokit.repos.uploadReleaseAsset({
-      data: fs.createReadStream(file),
-      headers: { 'content-length': size, 'content-type': 'application/octet-stream' },
-      name: path.basename(file),
-      url: release.data.upload_url,
-    });
+    runWithRetry(function () { uploadFile(file, size); });
   }
 }
 
-async function run() {
+async function uploadFile(file, size) {
+  core.info(`upload ${file}`);
+  await octokit.repos.uploadReleaseAsset({
+    data: fs.createReadStream(file),
+    headers: { 'content-length': size, 'content-type': 'application/octet-stream' },
+    name: path.basename(file),
+    url: release.data.upload_url,
+  });
+}
+
+async function runWithRetry(f) {
   const retries = 10;
+  const maxDelay = 120000;
+  let delay = 1000;
```

Comment:
`delay` seems to be unused?

Reply:
Yeah, don't mind that, the issue is that it fails in interesting ways: https://github.com/lnicola/rust-analyzer/runs/4574344966?check_suite_focus=true